### ISSUE TITLE
Python: minify('<style></style>', minify_css=True) hangs in forked process

### ISSUE THREAD

**Issue by bblanchon** [wilsonzlin/minify-html#26]
Hi,

Thank you for this excellent project!

I found that `minify('<style></style>', minify_css=True)` freeze child processes created by `os.fork()`.
Here is a  minimal repro ([see attached file for an extended version](https://github.com/wilsonzlin/minify-html/files/6122131/repro.py.txt))

```python
import os
from minify_html import minify

pid = os.fork()
minify('<style></style>', minify_css=False)
os.waitpid(pid, 0)
```

I tried both Python 3.7.10 and 3.8.8 running in the official Docker images based on debian:buster-slim.
**EDIT**: also on Python 3.8.5 on Ubuntu-20.04 running in WSL
**EDIT**: I forgot to mention that I used version 0.4.3 for all tests.

For the record, I discovered this bug while calling `minify()` from a [Celery](https://docs.celeryproject.org/en/v4.4.7/) task.
As a workaround, one can [select another pool implementation](https://docs.celeryproject.org/en/v4.4.7/) in Celery, such as `threads` or `solo`.

Best regards,
Benoit

**Comment by wilsonzlin** [comment]
Thanks for the details. This might be related to #29 but I haven't checked myself yet. Are you able to see if using the package [minify_html_core](https://pypi.org/project/minify-html-core/) solves the issue for you? It doesn't have the built-in JS/CSS minifier. If it does, it's probably caused by the same issue and will help me with coming up a solution.

**Comment by bblanchon** [comment]
@wilsonzlin, as I demonstrated in [the repro](https://github.com/wilsonzlin/minify-html/files/6122131/repro.py.txt), there is no problem when `minify_css=False`, so I don't think that testing with `minify_html_core` will bring any new information.

If I remember correctly, I also tested the JS minification, and it didn't have the issue either.
Only the CSS minification is affected.

**Comment by wilsonzlin** [comment]
I tested your provided repro code with `minify_html_core` and it exits successfully without hanging. Using `minify_html` does cause it to get stuck but does exit when it receives SIGINT.

`minify_html_core` is different because it doesn't include a statically-linked esbuild, which contains the cgo runtime. If you look at the issue I linked, the runtime has issues when coexisting with other code that handles signals. I suspected the same thing was happening here, which is why I suggested trying the core library which doesn't include esbuild.

As mentioned in the other issue, this problem is quite complex to solve as it is caused by the cgo runtime implementation, which is required to use any Go library in another language and isn't easily alterable. I'm investigating some potential solutions, like porting the esbuild transform API code to Rust. Another would be to switch to a different JS/CSS minification library usable from Rust, but there aren't many great options available AFAIK.
